### ISSUE TITLE
server crash

### ISSUE THREAD

**Issue by nasaboy** [EAimTY/tuic#174]
不知道为何server crash，日志见附件
client和server都是1.0.0-beta0
[crash.txt](https://github.com/EAimTY/tuic/files/11602290/crash.txt)

**Comment by ConnersHua** [comment]
同样遇到该问题，崩了几次了
```bash
May 30 22:52:17 IdolizedScratchy-VM tuic-server[1514175]: [2023-05-31T02:52:17Z WARN ] [0x9c076fb0] [221.239.151.151:64543] [unauthenticated] [authenticate] timeout
May 30 22:52:18 IdolizedScratchy-VM tuic-server[1514175]: thread 'tokio-runtime-worker' panicked at 'internal error: entered unreachable code', /project/tuic-server/src/connection/mod.rs:122:79
May 30 22:52:18 IdolizedScratchy-VM tuic-server[1514175]: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
May 30 22:52:18 IdolizedScratchy-VM systemd[1]: tuic.service: Main process exited, code=killed, status=6/ABRT
May 30 22:52:18 IdolizedScratchy-VM systemd[1]: tuic.service: Failed with result 'signal'.
May 30 22:52:18 IdolizedScratchy-VM systemd[1]: tuic.service: Consumed 4.694s CPU time.
```

**Comment by EAimTY** [comment]
should be fixed in 056f880a29b5c4af9dfc8d875a2f970e33256b84
